### PR TITLE
TarFetchedCallback and RouteTileDownloadListener: to kotlin

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

Migrate Java to Kotlin `TarFetchedCallback` and `RouteTileDownloadListener` [#issue](https://github.com/mapbox/navigation-sdks/issues/167)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+class TarFetchedCallbackTest {
+
+    @Test
+    fun onSuccessfulResponse_downloadTaskIsExecuted() {
+        val downloadTask = mockk<DownloadTask>(relaxed = true)
+        val callback = buildCallback(downloadTask)
+        val call = mockk<Call<ResponseBody>>()
+        val response = mockk<Response<ResponseBody>>()
+        val responseBody = mockk<ResponseBody>()
+
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.kt]: NIT no new line needed

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        verify { downloadTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, responseBody) }
+    }
+
+    @Test
+    fun onUnsuccessfulResponse_errorIsProvided() {
+        val downloader = mockk<RouteTileDownloader>(relaxed = true)
+        val callback = buildCallback(downloader)
+        val call = mockk<Call<ResponseBody>>()
+        val response = mockk<Response<ResponseBody>>(relaxed = true)
+
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.kt]: NIT no new line needed

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        verify { downloader.onError(any()) }
+    }
+
+    @Test
+    fun onFailure_errorIsProvided() {
+        val downloader = mockk<RouteTileDownloader>(relaxed = true)
+        val callback = buildCallback(downloader)
+        val call = mockk<Call<ResponseBody>>()
+        val throwable = mockk<Throwable>()
+
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.kt]: NIT no new line needed

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallback.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+internal class TarFetchedCallback(
+    private val downloader: RouteTileDownloader,
+    private val downloadTask: DownloadTask
+) : Callback<ResponseBody> {
+
+    override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
+        if (response.isSuccessful) {
+            downloadTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, response.body())
+        } else {
+            val errorCodes = HashMap<Int, String>()
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallback.kt]: 👀 https://github.com/mapbox/mapbox-navigation-android/pull/2099#discussion_r331742109

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallback.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            val errorMap = TarResponseErrorMap(errorCodes)
+            val error = OfflineError(errorMap.buildErrorMessageWith(response))
+            downloader.onError(error)
+        }
+    }
+
+    override fun onFailure(call: Call<ResponseBody>, throwable: Throwable) {
+        val error = OfflineError(throwable.message ?: "Unspecified error")
```
Comment (Lebedinsky) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallback.kt]: I think "Unspecified error" is not good, it is not informative at all. Would be good to put at least smth like "Tar fetching error" or "TarFetchedCallback onFailure called" or some another variant?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallback.kt (hunk 4 above)
Comment (RingerJK) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallback.kt]: left `FIXME` note

**Comment by RingerJK** [comment]
@Guardiola31337 @Lebedinsky PATL

**Pull request merged**